hasher-matcher-actioner/webapp/src/settings/ThreatExchangeSettingsTab.jsx

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ */
+
+import React, {useEffect, useState} from 'react';
+import {
+  Col,
+  Row,
+  Button,
+  Toast,
+  Form,
+  Spinner,
+  Tooltip,
+  OverlayTrigger,
+  Card,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import {CopyableTextField} from '../utils/TextFieldsUtils';
+import {
+  fetchAllDatasets,
+  syncAllDatasets,
+  udpateDataset,
+  deleteDataset,
+} from '../Api';
+
+export default function ThreatExchangeSettingsTab() {
+  const [datasets, setDatasets] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [syncing, setSyncing] = useState(true);
+  const [showToast, setShowToast] = useState(false);
+  const [toastBody, setToastBody] = useState(null);
+  const onPrivacyGroupSave = privacyGroup => {
+    udpateDataset(
+      privacyGroup.privacyGroupId,
+      privacyGroup.localFetcherActive,
+      privacyGroup.localWriteBack,
+    ).then(response => {
+      datasets[
+        datasets.findIndex(
+          item => item.privacy_group_id === response.privacy_group_id,
+        )
+      ] = response;
+      setDatasets(datasets);
+    });
+  };
+  const onPrivacyGroupDelete = privacyGroupId => {
+    deleteDataset(privacyGroupId).then(() => {
+      const filteredDatasets = datasets.filter(
+        item => item.privacy_group_id !== privacyGroupId,
+      );
+      setDatasets(filteredDatasets);
+    });
+  };
+  const onSync = () => {
+    syncAllDatasets().then(syncResponse => {
+      setSyncing(true);
+      if (syncResponse.response === 'Dataset is update-to-date') {
+        fetchAllDatasets().then(response => {
+          setDatasets(response.datasets_response);
+        });
+      } else {
+        alert('Errors when syncing privacy groups. Please try again later');
+      }
+    });
+  };
+  useEffect(() => {
+    fetchAllDatasets(setLoading(false)).then(response => {
+      setLoading(true);
+      setDatasets(response.datasets_response);
+    });
+  }, []);
+  return (
+    <>
+      <Card.Header>
+        <Row className="mt-3">
+          <h2 className="mt-2">ThreatExchange Privacy Groups </h2>
+          <OverlayTrigger
+            key="syncButton"
+            placement="right"
+            overlay={
+              <Tooltip id="tooltip-right">
+                Fetch privacy groups from ThreatExchange
+              </Tooltip>
+            }>
+            <Button
+              variant="primary"
+              onClick={() => {
+                setSyncing(false);
+                setToastBody('Privacy groups are up to date!');
+                setShowToast(true);
+                onSync();
```

Comment:
nit: All of this should be in the onSync() method.. 

No matter which component calls onSync(), these things would happen. Doing it in a function also allows you to later centralize common behaviors.